themes/bootstrap3/js/covers.js

```diff
@@ -2,22 +2,44 @@
 function loadCoverByElement(data, element) {
   var url = VuFind.path + '/AJAX/JSON?method=' + 'getRecordCover';
   var img = element.find('img');
-  var spinner = element.children('div.spinner');
-  var container = element.children('div.cover-container');
+  var spinner = element.find('div.spinner');
+  var container = element.find('div.cover-container');
+  var source = $('<p class="cover-source">' + VuFind.translate('cover_source_label') + ' </p>');
   function coverCallback(response) {
-    spinner.hide();
-    container.show();
     if (typeof response.data.url !== 'undefined' && response.data.url !== false) {
       img.attr("src", response.data.url);
-      container.children().not("img").hide();
+      var inlink = element.parent().is('a');
+      var medium = img.parents('.media-left, .media-right, .carousel-item');
+      if (typeof response.data.backlink_text !== 'undefined') {
+        if (typeof response.data.backlink_url !== 'undefined') {
+          var link = $('<a href="' + response.data.backlink_url + '" class="cover-backlink" target="_blank">' + response.data.backlink_text + '</a>');
+          source.append(link);
+        } else {
+          var span = $('<span class="cover-source-text"' + response.data.backlink_text + '</span>');
+          source.append(span);
+        }
+        if (inlink === true && response.data.backlink_on_results === true) {
```

Comment:
This seems a bit fragile -- in that the identification of whether the backlink is in results or not seems to be based on a variety of circumstances that might not be entirely specific. I wonder if we should just display backlinks everywhere and rely on CSS customization to hide them in unwanted places. Would that be simpler?

Reply:
Yes it is not the same (inlink vs search result). The reason I use this type of distinction is that, if a link is inserted into another link in html it simply does not work well and I need to make same dom manipulation to get the inner link out of outer link....

simple solution would be to rename the option to something like 'other the detail page'

or change the non-detail page logic somehow.... but then we may need much more options - channels, related tab, and possibly some more in future...

Reply:
We currently have cover size settings that can be configured based on template names: see [these configurations](https://github.com/vufind-org/vufind/blob/dev/config/vufind/config.ini#L864-L873). Perhaps we could use a similar mechanism for allowing the user to configure where backlinks appear. Maybe we could replace the mandatoryBacklinks boolean property with a mandatoryBacklinkLocations array, and then locations in this array could forcibly override user preferences for those locations (and when this is non-empty, that tells us we have to force AJAX covers).

I think that meets the use case and is flexible... but is it too complicated to implement? Should we go ahead with that approach, or should we do something simpler for now and refine as needed?

Reply:
To be honest, this seems a bit overcomplicated for our current need, although I agree it would be much more flexible, and I could imagine such a implementation in the future, if it is OK for you, I would like this to be merged as is and probably continue with this on another PR in the future.

Reply:
I agree that we don't need to fully implement this to get the PR merged, but I wonder if moving a little bit in that direction would make the code simpler and more readable. For example, what if we replace the boolean `isBacklinkMandatory()` and `showBacklinksOnResults()` methods with a single `mandatoryBacklinkLocations()` method that returns an array? Then current code that checks for the boolean can instead check for a non-empty array. For now, we can code this so that backlinks are only displayed when mandatory, which keeps other logic simple... but this makes it easier to add an additional configuration layer if we decide we need it in the future, and it also keeps the cover drivers more independent of the UI behavior -- they tell us details about the provider (where a backlink needs to be displayed, which is constant for each provider), but reconciling that with user preferences can be delegated to a higher layer of code. Does that make sense?

Reply:
That does make sense, I've done a bit of refactoring here.

Reply:
Thanks, this is definitely a step in the right direction! Would it be worth taking it a little bit farther and exposing the whole backlink_locations array to the JS instead of mapping it to the backlink_on_results boolean? Again, just trying to make this a little more generic if possible. I'm still not entirely comfortable with assuming things are in results just because of the presence of links around cover images, since I think there are some possible exceptions and edge cases there... but I'm willing to accept this as-is with a "TODO: make location detection more accurate/specific" comment if that's as far as you want to go right now!

Reply:
That makes sense @demiankatz. Now the whole backlink locations array is passed. I also tried to reuse the context for backlink display